docs/architecture/design-proposals/dashboard/application-view-revamp.md

```diff
@@ -0,0 +1,126 @@
+
+<img src="./img/pod-count.png" width="300px">
+
+### Highlight Application Credentials
+
+In the current view, application credentials can be found either reading and executing the commands that the Notes suggest, or clicking on the icon to reveal the secrets below. Those secrets usually are of the type "Opaque" (rather than `kubernetes.io/tls` or `kubernetes.io/service-account-token`). We can extract those credentials and show them in a more prominent place. That way the user can easily discover and copy the application credentials without the need of a terminal:
+
+<img src="./img/credentials-box.png" width="400px">
+
+Note that we will still be showing the different application Secrets along with all the other resources in its own table.
+
+### Compress Application Actions
+
+The current approach is to show one button per action (Upgrade, Delete, Rollback). This list is growing over time (for example, the backend endpoint for running tests is ready to be used). It can also be confusing to show a different list of buttons depending on the release state. One example is the Rollback button that is only rendered if the application has been upgraded. In order to avoid these issues, we can show a clickable menu with the different options, graying out the options not available (potentially showing a tooltip).
+
+We could also differentiate between the configuration and the ugprade actions. While both actions translate to the same helm command (`upgrade`), while configuring the application, the same version will be used and when upgrading, the latest version can be auto-selected.
+
+<img src="./img/configuration-options.png" width="300px">
+
+### Include Additional Information for Access URLs
+
+The current list of URLs can be improved with two small changes:
+
+- If the browser can access the URL, we could show an icon so the user knows if the URL is working.
+- If the URL is not working and the cause is well known, we could show additional information for the user to debug it (e.g. Pending IPs when using LoadBalancers in Minikube [link](https://github.com/kubeapps/kubeapps/issues/953)).
+
+ <img src="./img/url-list.png">
+
+### Render an Extended Resources Table
+
+Finally, we can show a single resource table with all the resources so users can inspect in detail the different resources of the application without adding too much noise to the view. Apart from the list of resources we are currently showing, we can add the list of pods related to the application (the ones that are related to a Deployment/Statefulset/Daemonset).
+
+Apart from the basic information of the resource, we could add a summary (human-friendly) to know the status of the resource if possible. 
+
+We can also show different buttons in order to show the resource YAML, description and logs if available (available for pods, services and workloads).
+
+<img src="./img/resources-table.png">
+
```

Comment:
"If we later develop a clearer way to capture and present errors to users, clearly identifying the cause without requiring clicking through resources, the resources table will still be useful as a way to dig deeper and learn more without leaving the UI."